### DISCUSSION TITLE
Prefix Redis keys

### DISCUSSION THREAD

**Discussion by pablote** [directus/directus#19741] [Feature Requests]
### Summary

It'd be nice if a Redis key prefix could be specified through configuration. When set, every key written and read from Redis would get this prefix attached to it.

Adding this as feature request cause looking at the config reference it didn't seem to exist.

### Basic Example

New config env var, eg:
`REDIS_KEY_PREFIX=myapp:`

This would cause a key like the current `namespace:system-cache_schema_shared` to be saved as `myapp:namespace:system-cache_schema_shared`

### Motivation

The motivation is for cases when using different Redis databases is not desired or possible. Is pretty easy to run out of redis database numbers, some hostings also don't allow multiple databases. This would also allow running n number of Directus instances pointing to the same Redis without risk of key collision.

### Detailed Design

Nothing more to add.

### Must Have:
- A way to set a key prefix through configuration
- Keys on Redis being prefixed based on said configuration

### Drawbacks

I can't think of any drawback. This would be a backward compatible change. Not setting the new config would have Directus bahave as it currently does, so no learning overhead.

### Alternatives

I'm not sure if there's any alternative to this, except for trying to handle this at the Redis level as much as possible. I can't think of a way to point 16+ Directus instances to the same Redis instance, that'd require multiple Redis processes/servers.

### Adoption Strategy

Not a breaking a change, current usage can remain as is.

**Comment by keesvanbemmel** [comment]
Also would make it possible to use 1 redis instance across multiple Directus instances. Or is that possible already?

**Comment by vs-johnny-huang** [comment]
I guess env `CACHE_NAMESPACE` is what you are looking for? (which is default to `system-cache`)

https://docs.directus.io/self-hosted/config-options.html#cache

**Comment by pablote** [comment]
Ahh. I just tested this and it does seem to do the trick. Somehow I missed that config setting, maybe because I was expecting it under REDIS_. I'll need to point multiple Directus instances to the same Redis to be fully sure this works and they don't collide at all. Thanks for pointing this out to me.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
